gnosis/cowswap
# [USD Price] Add coingecko api and hooks

# Summary

Adds Coingecko as a source for USD estimation on tokens to MAINNET and XDAI (no rinkeby or other test nets)

Currently it is part of a hook that get the GREATER of the 2 USD prices from USDC estimation vs Coingecko api resp. This can be discussed.

You can see in the logs the price found by coingecko:
![image](https://user-images.githubusercontent.com/21335563/131165135-8c9d595d-b6e4-4e47-940f-df2736032d7f.png)

  # To Test
1. On either Mainnet or Xdai try picking tokens and adding amounts
2. Check the networks tabs/console tabs for logs and info
3. check the FE for the amounts being displayed correctly
4. check xdai that more tokens have usd estimations (not that many unfort)

  # Next potentially
1. Add the A -> B -> A check mentioned by @anxolin as a fallback
2. Block swap button in some cases? maybe not part of this PR
  i. paraswap blocks swap button if price impact is -15%
  ii. if we get a returned USD price of -15% OR if we get no estimation and a bad A -> B -> A slippage then also block
